### ISSUE TITLE
keydown navigation in dropdown broken if there are hidden items

### ISSUE THREAD

**Issue by msh-odoo** [twbs/bootstrap#29405]
Bootstrap v4.3.1

When navigating on dropdown using keyboard and if there are hidden(display: none) dropdown item in between then keyboard navigation stops.

Issues seems from: ```_dataApiKeydownHandler``` method, it doesn't skip hidden element, it directly set focus on next element while next element can be hidden and focus does not set on hidden element and navigation gets broken.

**Comment by Johann-S** [comment]
Bug reports must include a live demo of the problem. Per our [contributing guidelines](https://github.com/twbs/bootstrap/blob/v4-dev/.github/CONTRIBUTING.md), please create a reduced test case via [CodePen](https://codepen.io/)/[JS Bin](https://jsbin.com/) or [Stackblitz](https://stackblitz.com/) and report back with your link, Bootstrap version, and specific browser and OS details.

*This is an automated reply*

**Comment by msh-odoo** [comment]
@Johann-S here is the CodePen link: https://codepen.io/mishekha/pen/ExYREam

Open dropdown and press Down key, focus goes to first item and then press Down key again, it doesn't go to next item in dropdown, it is because we have ```display: none``` item in between.

Bootstrap version I am using: 4.3.1, Browser Chrome: 76.0.3809.132 OS: macOS Mojave.

Hope this will help, let me know if you need further details.

**Comment by Johann-S** [comment]
it seems a normal behavior here 🤔 I think it's for an accessibility reason, maybe @patrickhlauke have some hints on that

**Comment by patrickhlauke** [comment]
@Johann-S that's broken behavior: if an item isn't there (as `display:none` effectively removes it both visually and from the accessibility tree), it should not receive focus (or attempt to receive focus). the script needs to be modified to be more discerning about only trying to attempt to go to items that are actually there...or @msh-odoo should remove items completely from the DOM tree, rather than just hiding them via CSS.

**Comment by Johann-S** [comment]
Hmm I see 🤔 we have a util helper to check `isVisible` we should use it

**Comment by msh-odoo** [comment]
@patrickhlauke @Johann-S I can not remove item from dropdown in my case, I did workaround fix to put all hidden items at last so that navigation works but it should be fixed in dropdown itself, before going to next item it should check whether that item is visible or not, if it is not visible then focus is not possible on that item so we should go for next item and so on.

**Comment by msh-odoo** [comment]
@Johann-S @patrickhlauke Any news on this?

**Comment by Johann-S** [comment]
It's totally open, so if anyone wants to add that, feel free to create a PR 👍

**Comment by rohit2sharma95** [comment]
Fixed in #29523

**Comment by rohit2sharma95** [comment]
Shouldn't this issue be closed? :thinking: @XhmikosR

**Comment by XhmikosR** [comment]
Yup. The PR that was merged didn't properly mention the issue hence why it remained open. I usually edit PR descriptions myself when they are wrong.

**Comment by msh-odoo** [comment]
Good news, Thanks @jeremyvii  for the fix
